src/averageAges.js

```diff
@@ -55,7 +75,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const callBackSon = (sum, child) =>
+    sum + (child.born
+      - people.find(mother => mother.name === child.mother).born);
+
+  const callBackFilter = (child) =>
+    onlyWithSon === true
+      ? people.some(mother => child.mother === mother.name
+          && child.sex === 'm')
+      : people.some(mother => child.mother === mother.name);
+
+  const children = people.filter(callBackFilter);
+  const result = children.reduce(callBackSon, 0) / children.length;
```

Comment:
You have repeated code `array.reduce(callBack, 0) / array.length`
Move this to helper function and reuse it